### DISCUSSION TITLE
Unknown arg `id` in data.model.connect

### DISCUSSION THREAD

**Discussion by ulisesvina** [prisma/prisma#13574] [Q&A]
I'm trying to create a following/followers system using Prisma, for my models I'm using the code over at the docs ([https://www.prisma.io/docs/concepts/components/prisma-schema/relations/self-relations#many-to-many-self-relations](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/self-relations#many-to-many-self-relations)), but when trying to update the data of the an user it returns the error `Unknown arg "id" in data.model.connect`.

For the client, I'm using the code below
```
    await prisma.user.update({
      where: { id: res.locals.user.id },
      data: {
        following: {
          connect: {
            id
          }
        }
      }
    });
```

**Comment by austincrim** [comment]
Hey @ulisesvina 👋 ,

I was able to run your query without an issue. 

- What version of Prisma are you on? 
- Do you have existing data in the `Users` table?

**Reply by ulisesvina** [reply]
Hello Austin, I'm running Prisma 3.14.0, and I have existing data in the Users table, but not on the user I'm trying to update. 

If needed, this is my Model schema:
```
model User {
  id                String    @id @default(cuid())
  username          String    @unique
  email             String    @unique
  bio               String
  password          String
  createdAt         DateTime  @default(now())
  Post              Post[]
  followers         Follows[] @relation(name: "following")
  following         Follows[] @relation(name: "follower")
}

model Follows {
  follower    User   @relation(name: "follower", fields: [followerId], references: [id])
  followerId  String
  following   User   @relation(name: "following", fields: [followingId], references: [id])
  followingId String

  @@id([followerId, followingId])
}
```

![image](https://user-images.githubusercontent.com/20508563/171267705-31d3fa6d-3f35-4a24-9647-6493f4cf3b16.png)

**Reply by austincrim** [reply]
That helps, thank you!

The `Follows` model does not have a field called `id` on it, which is why we are getting the type error with your query above. It is using a compound id of `followerId` and `followingId`. These are the fields we must use in our `connect` query.

```ts
await prisma.user.update({
  where: { id: 1 },
  data: {
    following: {
      connect: {
        followerId_followingId: {
          followerId: 'someId',
          followingId: 'someOtherId'
        }
      }
    }
  }
})
```

**Comment by ulisesvina** [comment]
Thanks a lot! That seemed to help, unfortunately I'm getting another error, it goes as follows:
`Expected 1 records to be connected after connect operation on one-to-many relation 'follower', found 0.`

Maybe this is a bug? As suggested by [the docs](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/self-relations#defining-multiple-self-relations-on-the-same-model) I'm trying to make a many-to-many relation, not a one-to-many.

**Reply by austincrim** [reply]
Could you share the code you're using? Are you passing existing userIds in your `connect`?

**Reply by ulisesvina** [reply]
Yes! I'm using existing user IDs, and this is the code I'm using:
```
    await prisma.user.update({
      where: { id },
      data: {
        following: {
          connect: {
            followerId_followingId: {
              followerId: id,
              followingId,
            },
          },
        },
      },
    });
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
